server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1035,7 +1046,7 @@ public IndexResult index(Index index) throws IOException {
                     assert index.origin().isFromTranslog() || indexResult.getSeqNo() == SequenceNumbers.UNASSIGNED_SEQ_NO;
                     localCheckpointTracker.markSeqNoAsPersisted(indexResult.getSeqNo());
                 }
-                indexResult.setTook(System.nanoTime() - index.startTime());
+                indexResult.setTook(relativeTimeInNanosSupplier.getAsLong() - index.startTime());
```

Comment:
I think this also degrades into 200ms granularity? This could provoke slow logs if slow logging is setup to warn about indexing taking longer than 200ms.